app/services/description_import.rb

```diff
@@ -89,4 +95,36 @@ def compact_params(params)
       params[key].map { |param| compact_params(param) }
     end
   end
+
+  def remove_contributors_if_role_without_name(compacted_params_hash)
+    return unless compacted_params_hash && compacted_params_hash[:contributor]
+
+    compacted_params_hash[:contributor].each do |contributor|
+      next unless contributor && contributor[:name].nil? && contributor[:role].present?
+
+      compacted_params_hash[:contributor].delete(contributor)
+    end
+  end
+
+  def remove_form_if_source_without_value(compacted_params_hash)
+    return unless compacted_params_hash && compacted_params_hash[:form]
+
+    compacted_params_hash[:form].each do |form|
+      next unless form && form[:value].nil? && (form[:source].present? || form[:type].present?)
+
+      compacted_params_hash[:form].delete(form)
+    end
+  end
+
+  def remove_nested_attributes_without_value(compacted_params_hash)
+    # event can have contributors, geographic can have form
```

Comment:
Should this comment also say something about `relatedResource`?

Reply:
done